plugins/modules/win_dns_zone.ps1

```diff
@@ -0,0 +1,231 @@
+#!powershell
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options = @{
+        name = @{ type = "str"; required = $true }
+        type = @{ type = "str"; choices = "primary", "secondary", "forwarder", "stub" }
+        replication = @{ type = "str"; choices = "forest", "domain", "legacy", "none" }
+        dynamic_update = @{ type = "str"; choices = "secure", "none", "nonsecureandsecure" }
+        state = @{ type = "str"; choices = "absent", "present"; default = "present" }
+        forwarder_timeout = @{ type = "int" }
+        dns_servers = @{ type = "list"; elements = "str" }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$check_mode = $module.CheckMode
+
+$name = $module.Params.name
+$type = $module.Params.type
+$replication = $module.Params.replication
+$dynamic_update = $module.Params.dynamic_update
+$state = $module.Params.state
+$dns_servers = $module.Params.dns_servers
+$forwarder_timeout = $module.Params.forwarder_timeout
+
+$parms = @{ name = $name }
+
+Function Get-DnsZoneObject {
+    Param([PSObject]$Object)
+    $parms = @{
+        name     = $Object.ZoneName.toLower()
+        type     = $Object.ZoneType.toLower()
+        paused   = $Object.IsPaused
+        shutdown = $Object.IsShutdown
+    }
+
+    if ($Object.DynamicUpdate) { $parms.dynamic_update = $Object.DynamicUpdate.toLower() }
+    if ($Object.IsReverseLookupZone) { $parms.reverse_lookup = $Object.IsReverseLookupZone }
+    if ($Object.ZoneType -like 'forwarder' ) { $parms.forwarder_timeout = $Object.ForwarderTimeout }
+    if ($Object.MasterServers) { $parms.dns_servers = $Object.MasterServers.IPAddressToString }
+    if (-not $Object.IsDsIntegrated) {
+        $parms.replication = "none"
+        $parms.zone_file = $Object.ZoneFile
+    } else {
+        $parms.replication = $Object.ReplicationScope.toLower()
+    }
+
+    return $parms | Sort-Object
+}
+
+Function Compare-DnsZone {
+    Param(
+        [PSObject]$Original,
+        [PSObject]$Updated)
+
+    if ($Original -eq $false) { return $false }
+    $props = @('ZoneType', 'DynamicUpdate', 'IsDsIntegrated', 'MasterServers', 'ForwarderTimeout', 'ReplicationScope')
+    $x = Compare-Object $Original $Updated -Property $props
+    if ($x.Count -eq 0) { return $true } else { return $false }
+}
+
+# attempt import of module
+Try { Import-Module DnsServer }
+Catch { $module.FailJson("The DnsServer module failed to load properly: $($_.Exception.Message)", $_) }
+
+Try {
+    # determine current zone state
+    $current_zone = Get-DnsServerZone -name $name
+    if ($check_mode) {
+        $module.Diff.before = Get-DnsZoneObject -Object $current_zone
```

Comment:
Why do you only do this when in `$check_mode`? I don't see `Diff.before` being set when not in check mode.

Reply:
This will be corrected, thank you!